drivers/led/sn32/rgb_matrix_sn32f24xb.c

```diff
@@ -548,6 +548,26 @@ void sn32f24xb_set_color_all(uint8_t r, uint8_t g, uint8_t b) {
     }
 }
 
+#if defined(SHARED_MATRIX) && defined(BOOTMAGIC_LITE)
+void matrix_init_kb(void) {
```

Comment:
using ` matrix_init_kb` outside of a keyboard directory is non standard and probably bad practice. A more sane approach would be to hijack the `bootmagic_lite` [declaration](https://github.com/SonixQMK/qmk_firmware/blob/sn32_develop/quantum/bootmagic/bootmagic_lite.c#L35) directly - this would probably be more reliable since bootmagic gets called before matrix functions. Code handling it might be a bit uglier however

Reply:
Ok, will try that approach. When making this I felt unsure what the agreed QMK convention for use of _kb, _custom methods are.  Looking through the existing keyboard seemed like many keyboards use matrix_init_kb as general purpose hardware initialization hook, not just initializing matrix. If that's a recommended usecase then it would indeed be wrong to use `matrix_init_kb` in the driver level.

I also tried using matrix_init_custom, but that didn't work because that's only called when CUSTOM_MATRIX is enabled. Currently sn32f24xb driver uses in my opinion somewhat hacky way of on hand being quite somewhat custom matrix, but instead of using CUSTOM_MATRIX=yes or lite, disabling parts of quantum/matrix.c with `if !defined(SHARED_MATRIX)` .  

Maybe it would be cleaner if the SHARED sn32f24 matrix was handled same as other CUSTOM_MATRIX?

In longer term it might be worth reconsidering  how matrix customization works.  It doesn't feel like current architecture encourages reusable non standard key matrix.  Maybe move it to similar driver pattern as other subystems?